hub-saml/src/main/java/uk/gov/ida/saml/core/security/AssertionsDecrypters.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ida.saml.core.security;
+
+import io.prometheus.client.Counter;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import uk.gov.ida.saml.security.AssertionDecrypter;
+import uk.gov.ida.saml.security.exception.SamlFailedToDecryptException;
+import uk.gov.ida.saml.security.validators.ValidatedResponse;
+import java.util.List;
+
+public class AssertionsDecrypters { 
+    private final List<AssertionDecrypter> assertionDecrypters;
+    
+    public AssertionsDecrypters(List<AssertionDecrypter> decrypters) {
+        this.assertionDecrypters = decrypters;
+    }
+    
+    public List<Assertion> decryptedAssertions(ValidatedResponse responseValidator, Counter counter, Class klass) {
```

Comment:
Instead if `klass`, could it be something more descriptive? I just had to go splunking through to work out what was expected here. Something like `ResponseValidatorKlass` maybe?

Reply:
Also, `responseValidator` param name is a bit confusing. Should it be validatedResponse?